starfish/test/codebook/test_metric_decode.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from starfish import Codebook, IntensityTable
+from starfish.types import Features, Indices, SpotAttributes
+
+
+def intensity_table_factory(data: np.ndarray=np.array([[[0, 3], [4, 0]]])) -> IntensityTable:
+    """IntensityTable with a single feature that was measured over 2 channels and 2 rounds."""
+
+    # generates spot attributes equal in size to the number of passed features.
+    # each attribute has coordinates (z, y, x) equal to the feature index, and radius 1.
+    spot_attributes_data = pd.DataFrame(
+        data=np.array([[i, i, i, 1] for i in np.arange(data.shape[0])]),
+        columns=[Indices.Z, Indices.Y, Indices.X, Features.SPOT_RADIUS]
+    )
+
+    intensity_table = IntensityTable.from_spot_data(data, SpotAttributes(spot_attributes_data))
+    return intensity_table
+
+
+def codebook_factory() -> Codebook:
+    """
+    Codebook with two codewords describing an experiment with three channels and two imaging rounds.
+    Both codes have two "on" channels.
+    """
+    codebook_array = [
```

Comment:
does it make sense to piggy back on `codebook_array_factory`?

Reply:
Yes. As you suggested, I have made a follow-up ticket to aggregate these factories into a single module: #526.

Reply:
Why can't I use :stately-ambrose: here? :(